packages/mermaid/src/diagrams/flowchart/parser/flow.jison

```diff
@@ -456,43 +486,43 @@ keywords
 
 
 textNoTags: textNoTagsToken
-    {$$=$1;}
+    {$$={text:$1, type: 'text'};}
     | textNoTags textNoTagsToken
-    {$$=$1+''+$2;}
+    {$$={text:$1.text+''+$2, type: $1.type};}
+    | STR
+    { $$={text: $1, type: 'text'};}
+    | MD_STR
+    { $$={text: $1, type: 'markdown'};}
     ;
 
 
-classDefStatement:CLASSDEF SPACE DEFAULT SPACE stylesOpt
+classDefStatement:CLASSDEF SPACE idString SPACE stylesOpt
```

Comment:
Seems that it has become a different language. It used to be literally `default` only, and suddenly it became an `idString`. Is that what we want?

Reply:
Yes, someone wrote a unit test that wanted default to appear by itself and be interpreted as a node. So, that's why `idString` has the `DEFAULT` token in its declaration. I had the choice of either removing that word and invalidating the unit test, or to remove this grammar statement and allowing that specific test to pass. From my understanding, we wouldn't want to do the same for every keyword because it would become unmanageable and probably cause unforeseen conflicts

Reply:
So, the second line used to allow the alphaNum statements, which weren't just alphanumeric characters but also symbols. idString serves the same purpose as the old alphaNum, however, it also contains the `DEFAULT` keyword to help pass a single unit test. This means that the two grammar statements conflict here given that idString both replaces alphanum and has the `DEFAULT` token in its declaration

Reply:
Really, I don't believe `DEFAULT` should even be in `idString`. I kept it in there because there is a unit test that wants the word `default` to be read as a node. That is the only keyword I believe can be read as a token. I don't think we should do this for all keywords because then it would lead to too many conflicts.

Reply:
After some thought, I think you're right. In this implementation, I think idString, and alphaNumStatement are near identical, but I left it to create a better conceptual separation of the two, because it would be confusing to use idString for everything. I will return the old grammar here.

Reply:
I see, you have combined two previous lines
```
CLASSDEF SPACE DEFAULT SPACE stylesOpt
CLASSDEF SPACE alphaNum SPACE stylesOpt
```
That is ok.

But in some places you simply replaced `alphaNum` with `idString`. And previously there was no a 'default' keyword. This means that we added `default` keyword to some constructions that previously had not contained it. Is that what we want?

- May be it is, and we must create a test case for that
- May be it is not, and it would be great to create a test case for that that shows that some graph definition was NOT allowed, and this behavior have not changed since then

Reply:
@nirname I added a test to showcase that idString should indeed be there. I think that the old way was a bit incorrect, because you were able to create a class called 'default' in `classDefStatement` but were not allowed to apply it because the `classStatement` expects an alphaNum which, as you pointed out, does not include the `DEFAULT` token. Again, I do not really think that we should be allowing any of the nodes or classes to be keywords but removing the `DEFAULT` keyword would technically be a breaking change.

Reply:
Could you drop a link to the test?

Reply:
When class def is default then it is applied to ALL the nodes, thus your node don't have to start with `default` word. So it is better to extend the test so that it would illustrate that the default style is applied independently of the node name. In this example it is applied to all the nodes
```
      graph TD

      classDef default fill:#a34,stroke:#000,stroke-width:4px,color:#fff 
      hello --> default
```

```mermaid
      graph TD

      classDef default fill:#a34,stroke:#000,stroke-width:4px,color:#fff 
      hello --> default
```
\
I have checked your concern about styles, you are absolutely right. We need a test for this as well. If it is legit to call a node `default` (and it is), we must be able to apply styles to the node called `default` (and we can in your PR), we need coverage.

```
flowchart TD
      classDef default fill:#a34
      hello --> default

      style default stroke:#000,stroke-width:4px
```

Reply:
long speech short:
* extend prev test
* add one more